Code/Mantid/Framework/DataHandling/src/LoadSpiceXML2DDet.cpp

```diff
@@ -193,6 +197,27 @@ void LoadSpiceXML2DDet::init() {
       new ArrayProperty<size_t>("DetectorGeometry"),
       "A size-2 unsigned integer array [X, Y] for detector geometry. "
       "Such that the detector contains X x Y pixels.");
+
+  declareProperty(
+      "LoadInstrument", true,
+      "Flag to load instrument to output workspace. "
+      "HFIR's HB3A will be loaded if InstrumentFileName is not specified.");
+
+  declareProperty(
+      new FileProperty("InstrumentFilename", "", FileProperty::OptionalLoad,
+                       ".xml"),
+      "The filename (including its full or relative path) of an instrument "
+      "definition file. The file extension must either be .xml or .XML when "
+      "specifying an instrument definition file. Note Filename or "
+      "InstrumentName must be specified but not both.");
```

Comment:
It is not an elegant solution at this moment.  This algorithm came along with HB3A's IDF.  Therefore, I kept the property 'InstrumentFilename' to test IDF that I generated from time to time.  As there is only one four-circle instrument (may in the world) that uses this type of XML file to record counts on 2D detector, therefore, I hard coded HB3A as the instrument.  If there is any other instrument that can be adapted to this algorithm, I will be happy to make the change.